sklearn/cluster/k_means_.py

```diff
@@ -911,7 +866,7 @@ class KMeans(BaseEstimator, ClusterMixin, TransformerMixin):
     """
 
     def __init__(self, n_clusters=8, init='k-means++', n_init=10,
-                 max_iter=300, tol=1e-4, precompute_distances='auto',
+                 max_iter=300, tol=1e-4, precompute_distances='not-used',
```

Comment:
why replacing the default 'auto'?

Reply:
the `precompute_distances` parameter won't be used any more with this PR. This implementation is faster than previous one and has low memory usage.